node/node.go

```diff
@@ -257,6 +262,9 @@ func New(host p2p.Host, consensus *bft.Consensus, db ethdb.Database) *Node {
 			node.AddStakingContractToPendingTransactions()
 			node.DepositToFakeAccounts()
 		}
+		if node.Role == BeaconLeader || node.Role == BeaconValidator {
```

Comment:
i'm planning to refactor the node.New

can you write a comment that we create this map to keep track of current stakes.

actually this data should be saved in somewhere in blockchain (block header?)

Reply:
This data is saved in the contract, but chao will explicitly put it in blockchain. Added a comment.